File: naTanjir/naTanjir.Model/Korisnici.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace naTanjir.Model
{
    public class Korisnici
    {
        public int KorisnikId { get; set; }

        public string Ime { get; set; } = null!;

        public string Prezime { get; set; } = null!;

        public string Email { get; set; } = null!;

        public string Telefon { get; set; } = null!;

        public string KorisnickoIme { get; set; } = null!;
        public byte[] Slika { get; set; }


        public bool IsDeleted { get; set; }

        public DateTime DatumRodjenja { get; set; }

        public virtual ICollection<KorisniciUloge> KorisniciUloges { get; set; }
        public int? RestoranId { get; set; }
        public virtual Restoran? Restoran { get; set; }


    }
}

File: naTanjir/UI/natanjir_mobile/lib/models/korisnici.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:natanjir_mobile/models/korisnik_uloga.dart';
import 'package:natanjir_mobile/models/restoran.dart';
part 'korisnici.g.dart';

@JsonSerializable()
class Korisnici {
  int? korisnikId;
  String? ime;
  String? prezime;
  String? email;
  String? telefon;
  String? korisnickoIme;
  String? datumRodjenja;
  String? slika;
  bool? isDeleted;
  int? restoranId;
  Restoran? restoran;
  List<KorisnikUloga>? korisniciUloges;

  Korisnici();

  factory Korisnici.fromJson(Map<String, dynamic> json) =>
      _$KorisniciFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$KorisniciToJson(this);
}
